tests/client.spec.js

```diff
@@ -279,6 +287,109 @@ describe('Client', () => {
       })
     })
   })
+
+  describe('queryTaxPayer', () => {
+    describe('when the taxpayer ID is invalid', () => {
+      beforeEach(() => {
+        nock('https://www.szamlazz.hu')
+          .post('/szamla/')
+          .replyWithFile(200, RESPONSE_FILE_PATHS.INVALID_TAXPAYER);
+      });
+  
+      it('should return taxpayerValidity as false', async () => {
+        const result = await client.queryTaxPayer(12345678);
+        
+        expect(result).to.deep.equal({
+          taxpayerValidity: false,
+        });
+      });
+    });
+
+    describe('when the taxpayer ID is valid', () => {
+      beforeEach(() => {
+        nock('https://www.szamlazz.hu')
+          .post('/szamla/')
+          .replyWithFile(200, RESPONSE_FILE_PATHS.VALID_TAXPAYER);
+      });
+  
+      it('should return correct taxpayer details', async () => {
+        const result = await client.queryTaxPayer(12345678);
+  
+        expect(result).to.deep.equal({
+          taxpayerValidity: true,
+          taxpayerId: '12345678',
+          vatCode: '2',
+          countyCode: '41',
+          taxpayerName: 'taxpayerName',
+          taxpayerShortName: 'taxpayerShortName',
+          address: {
+            countryCode: 'HU',
+            postalCode: '1000',
+            city: 'BUDAPEST',
+            streetName: 'TESZT',
+            publicPlaceCategory: 'UTCA',
+            number: '1.'
+          },
+        });
+      });
+    });
+  
+  })
+
+  describe('registerCreditEntry', () => {
+    describe('HTTP status', () => {
+      it('should handle failed requests', async () => {
+        nock('https://www.szamlazz.hu')
+          .post('/szamla/')
+          .reply(404)
+
+        await expect(client.registerCreditEntry(creditEntryOptions, [creditEntry])).rejectedWith('Request failed with status code 404')
+        nock.isDone()
+      })
+    })
+
+    describe('successful credit entry registration', () => {
+      beforeEach(() => {
+        nock('https://www.szamlazz.hu')
+          .post('/szamla/')
+          .replyWithFile(200, RESPONSE_FILE_PATHS.SUCCESS_WITHOUT_PDF, {
```

Comment:
the success response of the request is `xmlagentresponse=DONE` text. If I read well the documentation.